src/hugr/hugrmut.rs

```diff
@@ -294,11 +322,40 @@ where
             self.as_mut().op_types.set(new_node, nodetype.clone());
             let meta = other.get_metadata(node.into());
             self.as_mut()
-                .set_metadata(node.into(), meta.clone())
+                .set_metadata(new_node.into(), meta.clone())
```

Comment:
This was done in an earlier PR, no?

Reply:
Oh no, that was outline_cfg. but same error 😲

Reply:
Yep, seems like an easy to miss error :/
And we never do anything with the metadata, so the error is never triggered.